src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -202,9 +202,9 @@ void primitive_inst::update_impl() {
 void primitive_inst::check_memory_to_set(const memory& mem, const layout& layout) const {
     CLDNN_ERROR_LAYOUT_MISMATCH("network layout",
         "set memory layout",
-        mem.get_layout(),
+        mem.get_layout().get_layout_with_dims(),
         "expected layout",
-        layout,
+        layout.get_layout_with_dims(),
```

Comment:
same comment as above

Reply:
I reverted this commit (https://github.com/andrew-k-park/openvino/commit/664175c19211e1ab4e0e297e8784a33d6cc32750) and move partial shape comparison logic to equality operator in layout.